server/services/event.js

```diff
@@ -320,11 +335,25 @@ module.exports = class EventService {
         };
 
         for (let defender of defenders) {
-            await this.createPlayerEvent(gameId, gameTick, defender._id, this.EVENT_TYPES.PLAYER_COMBAT_CARRIER, data);
+            let defenderCombatResult = combatResult
+            defenderCombatResult.carriers = combatResult.carriers.map(c => {
+                if (c.specialist && c.specialist.name === 'Scrambler' && !defender._id.equals(c.ownedByPlayerId) && c.after !== 0) {
+                    return { ...c, before: '???', after: '???' }
+                }
+                return c
+            })
+            await this.createPlayerEvent(gameId, gameTick, defender._id, this.EVENT_TYPES.PLAYER_COMBAT_CARRIER, { ...data, combatResult: defenderCombatResult });
         }
 
         for (let attacker of attackers) {
-            await this.createPlayerEvent(gameId, gameTick, attacker._id, this.EVENT_TYPES.PLAYER_COMBAT_CARRIER, data);
+            let attackerCombatResult = combatResult
+            attackerCombatResult.carriers = combatResult.carriers.map(c => {
```

Comment:
There's a lot of duplication of code for masking scramblers, is it possible to move it into a shared function?

Reply:
Possible